capstone/src/test/java/com/google/sps/servlets/utils/ServletUtilTest.java

```diff
@@ -0,0 +1,98 @@
+package com.google.sps.servlets;
+
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+@RunWith(JUnit4.class)
+public final class ServletUtilTest {
```

Comment:
Ah sorry, I should've clarified. Util tests are really simple - since they have static methods, you can just call them directly so it requires zero set-up unlike a servlet. All you'd have to do is inside the tests, define the objects you are passing to the method, and then asserting that the result of calling the method on them is as you expect.

So something like

public void getPropertyList_getEmptyClubList() {
Entity entity = new Entity("Megha");
entity.setProperty(...);
....

assertEquals(ImmutableList.of(), ServletUtil.getPropertyList(entity, property));
}

Your current tests are still testing that, but it has a lot of unnecessary detail not relevant to the actual util method, which can make it harder to understand the tests and ensure they're testing exactly what the class/method is doing (in a sense, you're testing _too_ much). It still works, but isn't as ideal.

Reply:
Ah, I see this makes sense. I just removed the unnecessary statements / code. Thank you for bringing this up!